replit_river/session.py

```diff
@@ -491,9 +499,10 @@ async def close(self, is_unexpected_close: bool) -> None:
                 return
             self._state = SessionState.CLOSING
             self._reset_session_close_countdown()
-            await self.close_websocket(self._ws_wrapper, should_retry=False)
+            async with self._ws_lock:
+                await self._ws_wrapper.close()
```

Comment:
I notice in WebsocketWrapper.close() that it sets `ws_state` to `CLOSING` and then `CLOSED`, but no one would ever observe `CLOSING` because there's no `await` call between the two states. We don't wait for the websocket to actually close. I think that's fine, but FYI.

Reply:
very true

Reply:
i think this is fine for now, i don't see us want to observe closed at any degree.